File: Q09/Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int arr[] = new int[n];

        for(int i=0;i<n;i++){
            arr[i]=sc.nextInt();
        }

        for(int i=0;i<n;i++){
            if(arr[i]>=0)
                continue;
            int j=i;
            while(j>0 && arr[j-1]>=0){
                int temp=arr[j];
                arr[j]=arr[j-1];
                arr[j-1]=temp;
                j--;
            }
        }
        for(int i=0;i<n;i++){
            System.out.print(arr[i]+" ");
        }
    }    
}

File: Q09/main.py
n=int(input())
arr=[]
for i in range(n):
    arr.append(int(input()))

for i in range(n):
    if arr[i]>=0:
        continue
    j=i
    while(j>0 and arr[j-1]>=0):
        temp = arr[j]
        arr[j] = arr[j-1]
        arr[j-1]=temp
        j-=1

for i in range(n):
    print(arr[i],end=" ")